app/controllers/reviews_controller.rb

```diff
@@ -1,3 +1,46 @@
 class ReviewsController < ApplicationController
+  before_action :authenticate_user!
 
+  def edit
+    @product = Product.find(params[:product_id])
+    @review = @product.reviews.find(params[:id])
+  end
+
+  def create
+    @product = Product.find(params[:product_id])
+    @review = @product.reviews.new(review_params)
+    @review.user = current_user
+    @review.save
+    redirect_to product_path(@product)
+    # if @review.save
```

Comment:
kindly handle this exception here. if any validation fails in saving a review user should see a error message